### DISCUSSION TITLE
Route 53 AWS DNS provider question.

### DISCUSSION THREAD

**Discussion by agustinsilvano** [polymind-inc/acmebot#290] [Q&A] [answered]
Hey!

I want to ask about the access claims required to set Route 53 AWS as a DNS provider.

I saw the code [here](https://github.com/shibayan/keyvault-cmebot/blob/master/KeyVault.Acmebot/Providers/Route53Provider.cs) and the unique interaction with the DNS provider are the creation and deletion of the TXT RRs.

There is another interaction with the DNS provider somewhere?

I just want to know the impact to place the AWS credentials into the function.

Awesome tool.

Thanks.

Agustín.

**Comment by shibayan** [comment, marked as answer]
Be sure to create and use an IAM user with limited privileges. You can refer to the certbot documentation for help.

https://certbot-dns-route53.readthedocs.io/en/stable/#sample-aws-policy-json

I have tried using a policy that only allows operations of Route 53. You can refine your resources as needed.

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": "route53:*",
            "Resource": "*"
        }
    ]
}
```
